torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -319,7 +319,7 @@ class TestDistBackend(MultiProcessTestCase):
     def setUpClass(cls):
         os.environ["MASTER_ADDR"] = str(MASTER_ADDR)
         os.environ["MASTER_PORT"] = str(MASTER_PORT)
-        # os.environ["WORLD_SIZE"] = str(WORLD_SIZE)
+        os.environ["NCCL_ASYNC_ERROR_HANDLING"] = "1"
```

Comment:
We should add a comment here that NCCL_BLOCKING_WAIT overrides this and as a result tests like `test_batch_isend_irecv_nccl` would be fine.

Btw are there no tests that currently enable `NCCL_ASYNC_ERROR_HANDLING`? I couldn't find any test files enabling this. @osalpekar Should we add some tests for that verify `NCCL_ASYNC_ERROR_HANDLING` behavior?

Reply:
There's no unit tests AFAIK, but I think there is an internal mechanism for testing that it works appropriately

Reply:
Sure I can add some unit tests for NCCL Async Error Handling. We have a pretty useful flow for verifying correctness which is more of an integration test since it tests torchelastic + Async Error Handling e2e, but a unit tests would be good to have as well.